nodebalancer_types.go

```diff
@@ -0,0 +1,45 @@
+package linodego
+
+import (
+	"context"
+)
+
+// NodeBalancerType represents a single valid NodeBalancer type.
+type NodeBalancerType struct {
+	baseType[NodeBalancerTypePrice, NodeBalancerTypeRegionPrice]
+}
+
+// NodeBalancerTypePrice represents the base hourly and monthly prices
+// for a NodeBalancer type entry.
+type NodeBalancerTypePrice struct {
```

Comment:
I gave each resource type its own price and region prices structs so external users can still access them separate from their parent struct.

Reply:
I learned this pattern and put them similarly in python-sdk as well.